valhalla/jawn/src/managers/stripe/StripeManager.ts

```diff
@@ -1228,71 +1224,60 @@ WHERE (${builtFilter.filter})`,
     }
   }
 
-  public async reportUsageToStripe(
-    customerId: string,
-    usage: number
-  ): Promise<Result<null, string>> {
-    try {
-      // Assuming you have a usage item ID for each customer
-      const usageRecordParams: Stripe.SubscriptionItemCreateUsageRecordParams =
-        {
-          quantity: usage,
-          timestamp: Math.floor(Date.now() / 1000),
-          action: "set",
-        };
-
-      await this.stripe.subscriptionItems.createUsageRecord(
-        "si_1234", // Replace with actual subscription item ID
-        usageRecordParams
-      );
-
-      return ok(null);
-    } catch (error: any) {
-      return err(`Error reporting usage to Stripe: ${error.message}`);
-    }
-  }
-
-  public async handleStripeWebhook(
-    body: any,
-    signature: string
-  ): Promise<Result<null, string>> {
+  public async createCloudGatewayCheckoutSession(
+    origin: string,
+    amount: number,
+  ): Promise<Result<string, string>> {
     try {
-      const event = this.stripe.webhooks.constructEvent(
-        body,
-        signature,
-        process.env.STRIPE_WEBHOOK_SECRET!
-      );
-
-      if (event.type === "invoice.created") {
-        const invoice = event.data.object as Stripe.Invoice;
-        console.log("Invoice created:", invoice.id);
-        // Add your logic here to process the invoice
+      const customerId = await this.getOrCreateStripeCustomer();
+      if (customerId.error || !customerId.data) {
+        return err("Error getting or creating stripe customer");
       }
 
-      if (event.type === "invoice.upcoming") {
-        const invoice = event.data.object as Stripe.Invoice;
-        const customerId = invoice.customer as string;
+      const tokenUsageProductId = process.env.STRIPE_CLOUD_GATEWAY_TOKEN_USAGE_PRODUCT;
```

Comment:
we have something called `GET_KEY` and we have `SettingsManager` (Which we need to dry up) 

But can we use this so we can add this product ID in settings?